File: Week2/1_fibonacci_number/fibonacci.cpp
#include <iostream>

using namespace std;
int fibonacci_naive(int n){
    if (n <=1)
        return n;
    else
        return fibonacci_naive(n - 1) + fibonacci_naive(n - 2);
}

int fibonacci_fast(int n){
int f[n+2];
int i;

f[0] = 0;
f[1] = 1;

for (i = 2; i <= n; i++)
{
    f[i] = f[i-1] + f[i-2];
}
return f[n];

}

int main(){

int n = 0;
cout << "Enter the number you want to calculate fibonacci: ",
cin >> n;

//cout << fibonacci_naive(n) << '\n';
cout << fibonacci_fast(n) << '\n';
return 0;
}

File: Week2/1_fibonacci_number/fibonacci.py
def fibonacci_naive (n):
    if (n <= 1):
        return n
    else:
        return (fibonacci_naive(n-1) + fibonacci_naive(n-2))

def fibonacci_fast(n):
    f = []
    f.insert(0, 0)
    f.insert(1, 1)

    for i in range(2, n+1, 1):
        f.insert(i, (f[i-1] + f[i-2]))

    return f[n]


if __name__ == '__main__':
    
    n = int(input("Enter the fibonacci you want to know:"))
    print(fibonacci_naive(n))
    print(fibonacci_fast(n))